packages/jsii-pacmak/lib/targets/go.ts

```diff
@@ -27,12 +33,86 @@ export class Golang extends Target {
    * @param outDir    the directory where the publishable artifact should be placed.
    */
   public async build(sourceDir: string, outDir: string): Promise<void> {
-    return this.copyFiles(sourceDir, outDir);
+    // copy generated sources to the output directory
+    await this.copyFiles(sourceDir, outDir);
+
+    const pkgDir = path.join(outDir, goPackageName(this.assembly.name));
+
+    // write `local.go.mod` with "replace" directives for local modules
+    const localGoMod = await this.writeLocalGoMod(pkgDir);
+
+    // run `go build` with local.go.mod
+    await go('build', ['-modfile', localGoMod], { cwd: pkgDir });
+
+    // delete local.go.mod from the output directory so it doesn't get published
+    await fs.unlink(path.join(pkgDir, localGoMod));
+  }
+
+  /**
+   * Creates a copy of the `go.mod` file called `local.go.mod` with added
+   * `replace` directives for local mono-repo dependencies. This is required in
+   * order to run `go fmt` and `go build`.
+   *
+   * @param pkgDir The directory which contains the generated go code
+   */
+  private async writeLocalGoMod(pkgDir: string) {
+    const replace: Record<string, string> = {};
+
+    // find local deps by check if `<jsii.outdir>/go` exists for dependencies
+    // and also consider `outDir` in case pacmak is executed using `--outdir
+    // --recurse` (in which case all go code will be generated there).
+    const dirs = [
+      path.dirname(pkgDir),
+      ...(await findLocalBuildDirs(this.packageDir, 'go')),
+    ];
+
+    // try to resolve @jsii/go-runtime (only exists as a devDependency)
+    const jsiiRuntimeDir = tryFindLocalRuntime();
+    if (jsiiRuntimeDir) {
+      replace[JSII_RT_MODULE_NAME] = jsiiRuntimeDir;
+    }
+
+    // iterate (recursively) on all package dependencies and check if we have a
+    // local build directory for this module. if
+    // we do, add a "replace" directive to point to it instead of download from
+    // the network.
+    const visit = (pkg: RootPackage) => {
+      for (const dep of pkg.packageDependencies) {
+        for (const baseDir of dirs) {
+          const moduleDir = tryFindLocalModule(baseDir, dep);
+          if (moduleDir) {
+            replace[dep.goModuleName] = moduleDir;
+          }
+        }
+
+        // recurse to transitive deps ("replace" is only considered at the top level go.mod)
+        visit(dep);
+      }
+    };
+
+    visit(this.goGenerator.rootPackage);
+
+    // write `local.go.mod`
+
+    // read existing content
+    const goMod = path.join(pkgDir, GOMOD_FILENAME);
+    const lines = (await fs.readFile(goMod, 'utf-8')).split('\n');
+
+    for (const [from, to] of Object.entries(replace)) {
+      logging.info(`Local replace: ${from} => ${to}`);
+      lines.push(`replace ${from} => ${to}`);
```

Comment:
Somehow I thought you needed to have the version in the `replace` directive (`replace pkg version => localPath`), and I'm kinda glad I was wrong :)

Reply:
No need for local "replace"